Source/WebCore/bindings/js/JSDOMConvertPromise.h

```diff
@@ -89,4 +91,25 @@ template<typename T> struct JSConverter<IDLPromise<T>> {
     }
 };
 
+template<typename T> struct JSConverter<IDLPromiseIgnoringSuspension<T>> {
```

Comment:
We could have this derive from `JSConverter<IDLPromise<T>>` so we can share `needsState`, `needsGlobalObject`, and the `convert` template that takes a promise proxy, to make this a little less repetitive.

I also think that the `convert` function that takes `RefPtr<DOMPromise>` should probably be changed to take a `const RefPtr<DOMPromise>&` instead to avoid a bit of reference count churn.